lib/features/route_map/views/route_map_view.dart

```diff
@@ -54,47 +39,40 @@ class RouteMapViewState extends ConsumerState<RouteMapView> {
       }
     });
 
+    final landmarks = route?.landmarks ?? IList();
+    final visitedCount = landmarks.isEmpty ? 0 : (landmarks.length <= 2 ? landmarks.length : 2);
+
     return Scaffold(
+      appBar:
+          route == null
+              ? AppBar(
+                title: Text(context.l10n.choose_route),
+                foregroundColor: context.colorScheme.onPrimary,
+                backgroundColor: context.colorScheme.primary,
+              )
+              : null,
       body: Stack(
         children: [
-          RouteMapWidget(
-            landmarks: widget.route.landmarks,
-            visitedCount: 3,
-            active: _currentSheetState == SheetState.hidden,
-          ),
-          RouteProgressBar(landmarks: widget.route.landmarks, visitedCount: 3),
-          MapBottomSheet(
-            button: MainActionButton(
-              text: context.l10n.end_route,
-              backgroundColor: context.colorScheme.error,
-              onPressed: () async {
-                ref.read(sheetTriggerProvider.notifier).state = true;
-                await showDialog<EndRouteModal>(context: context, builder: (context) => const EndRouteModal());
-              },
+          if (route != null) ...[
+            RouteMapWidget(
+              landmarks: route.landmarks,
+              visitedCount: visitedCount,
+              active: _currentSheetState == SheetState.hidden,
             ),
-            controls: Row(
-              spacing: BottomSheetHeaderConfig.controlsSpacing,
-              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-              children: [
-                SecondaryActionButton(
-                  onPressed: () => ref.read(sheetModeProvider.notifier).state = SheetMode.half,
-                  text: context.l10n.route_description,
-                ),
-
-                SecondaryActionButton(
-                  onPressed: () => ref.read(sheetModeProvider.notifier).state = SheetMode.expanded,
-                  text: context.l10n.playlist,
-                ),
-              ],
-            ),
-
-            child:
-                (_currentSheetMode == SheetMode.half)
-                    ? const RouteInfoSection()
-                    : PlaylistInfoSection(songs: mockSongs),
-          ),
+            RouteProgressBar(landmarks: route.landmarks, visitedCount: 3),
+            const RouteBottomSheet(),
+          ] else ...[
+            RouteMapWidget(active: _currentSheetState == SheetState.hidden),
+            ChooseRouteBottomSheet(),
+          ],
         ],
       ),
     );
   }
+
+  @override
+  void dispose() {
+    ref.read(routeProvider.notifier).state = null;
```

Comment:
when in route details view you press on End Route button and confirm ending the route, the error is thrown here:
StateError (Bad state: Cannot use "ref" after the widget was disposed.)

Reply:
same happens when  you exit of the select route view using the back arrow